signer/src/transaction_coordinator.rs

```diff
@@ -305,6 +316,28 @@ where
             };
         }
 
+        // If we have just deployed the smart contracts, trying to broadcast the
+        // key rotation tx would result in transaction rejected, reason: NoSuchContract
+        if self.sbtc_contracts_deployed && !just_deployed_contracts {
```

Comment:
I think I would this check should be a little different. I think the check should be
1. whether the contract has been deployed and
2. whether the current aggregate key on the stacks node is the same as the latest one in the database.

The (2) matches what the signers should do during validation.

Reply:
Addressed in [0febb85](https://github.com/stacks-network/sbtc/pull/788/commits/0febb856000e877689aac10d7cb00002b1eff071)